src/main/java/Deadline.java

```diff
@@ -0,0 +1,26 @@
+public class Deadline extends Task {
+    protected String by;
+
+    public Deadline(String description, int taskID) {
+        super(description, taskID);
+        if (super.description.contains("/by")) {
+            int indexOfBy = super.description.indexOf("/by");
+            if (indexOfBy + 3 < super.description.length()) {
+                by = super.description.substring(indexOfBy + 3).trim();
+                this.description = super.description.substring(0, indexOfBy);
+            }
+            else {
```

Comment:
You may want to move the else clause up to the previous line to comply with the coding standard.